lib/server.go

```diff
@@ -252,33 +254,24 @@ func (srv *Server) BroadcastTransaction(txn *MsgDeSoTxn) ([]*MempoolTx, error) {
 		return nil, errors.Wrapf(err, "BroadcastTransaction: ")
 	}
```

Comment:
Ugh it deleted this long comment I wrote and now I have to type it again... I was saying that this could break frontends because they have the following flow:
1. Backend constructs txn and sends it to frontend to sign
2. Frontend signs and sends to /submit-transaction
3. /submit-transaction calls BroadcastTransaction (this thing) and returns
4. Then the frontend refreshes under the assumption the txn is in the mempool and its effect will show up after the refresh.

If we delete this thing then it could cause frontends to miss the txn's effect on refresh. A few ways to fix this:
1. Have frontend wait for inclusion in a block, which will take a bit of time. Not a big deal. We could wait for a post but maybe not a diamond or things like that.
2. Alternatively we could be smarter about how we render things in the mempool in the frontend.

I don't think we have to worry about this for now. I'm just mentioning for when you encounter it later.

Reply:
You're right, this totally breaks the frontends. I'm seeing it already on regtest. cc: @lazynina since this is something we'll need to solve.

Past discussions on this landed on optimistic UI updates for less critical and idempotent txns (posts, comments, likes,...) and to wait for block confirmation for more sensitive txns (basic transfers, limit order,...). The state syncer was the natural place to do optimistic updates. As a v0 solution though, maybe it's fine to wait for one block conf for everything cc: @lazynina so you're aware